src/Microsoft.SqlTools.ServiceLayer/Utility/SqlScriptFormatter.cs

```diff
@@ -176,6 +176,83 @@ public static string UnwrapLiteral(string literal)
             return literal;
         }
 
+        public static string[] DecodeMultipartIdenfitier(string multipartIdentifier)
+        {
+            StringBuilder sb = new StringBuilder();
+            List<string> namedParts = new List<string>();
+            bool insideBrackets = false;
+            bool bracketsClosed = false;
+            for (int i = 0; i < multipartIdentifier.Length; i++)
+            {
+                char iChar = multipartIdentifier[i];
+                if (insideBrackets)
+                {
+                    if (iChar == ']')
+                    {
+                        if (HasNextCharacter(multipartIdentifier, ']', i))
+                        {
+                            // This is an escaped ]
+                            sb.Append(iChar);
+                            i++;
+                        }
+                        else
+                        {
+                            // This bracket closes the bracket we were in
+                            insideBrackets = false;
+                            bracketsClosed = true;
+                        }
+                    }
+                    else
+                    {
+                        // This is a standard character
+                        sb.Append(iChar);
+                    }
+                }
+                else
+                {
+                    switch (iChar)
+                    {
+                        case '[':
+                            if (bracketsClosed)
+                            {
+                                throw new FormatException();
```

Comment:
A message might be good here, considering this is a pretty specific syntax error.